### PR TITLE
Added e2e test for `site.changed` webhook event

### PR DESCRIPTION
**Pull request by iamhalldor**
#15537 
There's a lot of API calls that can trigger a `site.changed` webhook event. I just cherry-picked deleting a post. If there's a more generic way to write this test I'm all ears.

### CODE REVIEW

File: ghost/core/test/e2e-webhooks/site.test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    title: 'webhookz',
+                    status: 'published',
+                    mobiledoc: fixtureManager.get('posts', 1).mobiledoc
+                }]
+            })
+            .expectStatus(201);
+
+        const id = res.body.posts[0].id;
+
+        await adminAPIAgent
```
Comment (naz) [inline ghost/core/test/e2e-webhooks/site.test.js:43]: Deleting a post is not necessary here. Publishing a post with a title "webhookz" doest the job too. Less code is better code 😊 

The 'site.changed' event is tiggered when Ghost [sends a cache invalidation header](https://github.com/TryGhost/Ghost/blob/ab598dd7ea3d1cc82f80bec5d0d04780a7edfe2e/ghost/core/core/server/web/parent/middleware/emit-events.js#L8-L10), which is triggered when [the post is published](https://github.com/TryGhost/Ghost/blob/ab598dd7ea3d1cc82f80bec5d0d04780a7edfe2e/ghost/core/core/server/api/endpoints/posts.js#L123-L127).

**Review by naz — CHANGES_REQUESTED** [review]
A minor change suggestion regarding triggering 'site.changed' event. The rest looks awesome! Thanks @halldorbjarni 🙌

File: ghost/core/test/e2e-webhooks/site.test.js (hunk 1 above)
Comment (halldorbjarni) [reply ghost/core/test/e2e-webhooks/site.test.js]: Ah good catch, I missed this. Should be fixed now.

**Comment by halldorbjarni** [comment]
Updated to only upload a single post. I don't see a squash and merge option, @naz do you want me to squash locally and push or is this something maintainers can do from the UI when merging?

**Comment by ErisDS** [comment]
Hey @halldorbjarni, we can squash our side so no worries about that 🙂 we'll get this reviewed again ASAP.

**Review by naz — APPROVED** [review]
Nice, this is great. Thanks @halldorbjarni 🙌

**Comment by ErisDS** [comment]
Hi @halldorbjarni thanks so much for creating and taking the time to update this PR for us.

It has now been merged 🎉 and will appear in the next release of Ghost - usually Fridays.

I'm not sure if you found this through hacktoberfest, but I've added the `hacktoberfest-accepted` label to this PR to make sure it counts. 

We have many open issues and ongoing projects for hacktoberfest and beyond. We would love to see you contribute some more ❤️

**Pull request merged**